### PR TITLE
[Bugfix] #7452 Invite button should be unclickable until user selected the friends to invite

### PR DESCRIPTION
**Pull request by jaworovcki**
Issue: [link](https://github.com/ita-social-projects/GreenCity/issues/7452)

### CODE REVIEW

File: src/app/main/component/user/components/habit/add-new-habit/habit-invite-friends/habit-invite-friends-pop-up/habit-invite-friends-pop-up.component.html [hunk 1]
```diff
@@ -67,7 +67,7 @@
     <button class="btn btn-outline-success" mat-dialog-close>
       {{ 'user.habit.invite.btn.cancel' | translate }}
     </button>
-    <button class="btn btn-success" [mat-dialog-close]="true" cdkFocusInitial (click)="setAddedFriends()">
+    <button class="btn btn-success" [mat-dialog-close]="true" cdkFocusInitial (click)="setAddedFriends()" [disabled]="!isAnyFriendSelected">
```
Comment (Lokankara) [inline src/app/main/component/user/components/habit/add-new-habit/habit-invite-friends/habit-invite-friends-pop-up/habit-invite-friends-pop-up.component.html]: Attribute cdkFocusInitial is not allowed here

File: src/app/main/component/user/components/habit/add-new-habit/habit-invite-friends/habit-invite-friends-pop-up/habit-invite-friends-pop-up.component.ts [hunk 2]
```diff
@@ -26,6 +26,10 @@ export class HabitInviteFriendsPopUpComponent implements OnInit, OnDestroy {
     private localStorageService: LocalStorageService
   ) {}
 
+  get isAnyFriendSelected(): boolean {
+    return this.friends && this.friends.some((friend) => friend.added);
```
Comment (Lokankara) [inline src/app/main/component/user/components/habit/add-new-habit/habit-invite-friends/habit-invite-friends-pop-up/habit-invite-friends-pop-up.component.ts]: You can use an optional chain like this: ` return this.friends?.some(friend => friend.added) ?? false;`

Comment (jaworovcki) [reply src/app/main/component/user/components/habit/add-new-habit/habit-invite-friends/habit-invite-friends-pop-up/habit-invite-friends-pop-up.component.ts]: update

File: src/app/main/component/user/components/habit/add-new-habit/habit-invite-friends/habit-invite-friends-pop-up/habit-invite-friends-pop-up.component.html (hunk 1 above)
Comment (jaworovcki) [reply src/app/main/component/user/components/habit/add-new-habit/habit-invite-friends/habit-invite-friends-pop-up/habit-invite-friends-pop-up.component.html]: update

**Review by kryzanivska-nastya — APPROVED** [review]

**Review by hnativlyubomyr — APPROVED** [review]

**Review by Lokankara — APPROVED** [review]

**Pull request merged**